snuba/consumer.py

```diff
@@ -4,45 +4,56 @@
 
 from batching_kafka_consumer import AbstractBatchWorker
 
-from . import processor
-from .writer import row_from_processed_event, write_rows
+from .writer import write_rows
+from .processor import Processor
 
 
 logger = logging.getLogger('snuba.consumer')
 
 
-class InvalidActionType(Exception):
-    pass
-
-
 class ConsumerWorker(AbstractBatchWorker):
-    def __init__(self, clickhouse, dist_table_name, producer, replacements_topic, metrics=None):
+    """
+    ConsumerWorker processes the raw stream of events coming
+    from Kafka, validates that they have the correct format
+    and does one of 2 things:
+
+        - If they are INSERTs, creates insertable rows from them
+          and batches and inserts those rows to clickhouse.
+        - If they are REPLACEs, forwards them to the replacements
+          Kafka topic, so that the Replacements consumer can deal
+          with them.
+    """
+    def __init__(self, clickhouse, dataset, producer, replacements_topic, metrics=None):
         self.clickhouse = clickhouse
-        self.dist_table_name = dist_table_name
+        self.dataset = dataset
         self.producer = producer
         self.replacements_topic = replacements_topic
         self.metrics = metrics
 
     def process_message(self, message):
         value = json.loads(message.value())
 
-        processed = processor.process_message(value)
-        if processed is None:
+        validated = self.dataset.PROCESSOR.validate_message(value)
+        if validated is None:
```

Comment:
Is there going to be a difference between the validator returning a None event and throwing an exception for an invalid event?
Also is it worth logging this case? Or do we expect the dataset processor to decide how to alert people of invalid messages?